web/packages/teleport/src/stores/storeUserContext.ts

```diff
@@ -167,6 +167,6 @@ export default class StoreUserContext extends Store<UserContext> {
   }
 
   hasDiscoverAccess() {
-    return this.hasPrereqAccessToAddAgents() || this.hasAccessToQueryAgent();
+    return this.hasPrereqAccessToAddAgents() && this.hasAccessToQueryAgent();
```

Comment:
@ryanclark i vaguely remember someone discussing this, it should've been an AND, but was wondering if there was a reason it was an OR?

Reply:
oh wait a minute, this flag is to show the discover feature on our dashboard. i think this is correct (or it's okay to show it), it's a way to "advertise" this feature if you have partial permissions. user should still be able to see it, and we let the user know possible reasons why they can't continue the flow

Reply:
ok, I'll close then

Reply:
From an admin perspective I wouldn’t want my users knowing they could add resources themselves even though they actually can’t. I’d be fine with default is we show it but we should have an option to hide it like we do other UI elements. We shouldn’t advertise what you can’t do IMO.